src/simulator/SimulatorActionBar.tsx

```diff
@@ -4,38 +4,70 @@
  * SPDX-License-Identifier: MIT
  */
 import { BoxProps, HStack, IconButton } from "@chakra-ui/react";
-import { useCallback } from "react";
-import { RiPlayFill, RiStopFill } from "react-icons/ri";
-import { useSimulator } from "../device/device-hooks";
+import { useCallback, useEffect, useState } from "react";
+import { RiPlayFill, RiRefreshLine, RiStopFill } from "react-icons/ri";
+import {
+  SyncStatus,
+  useSimulator,
+  useSyncStatus,
+} from "../device/device-hooks";
 import { useFileSystem } from "../fs/fs-hooks";
 
 interface SimulatorActionBarProps extends BoxProps {}
 
+enum SimState {
+  RUNNING,
+  STOPPED,
+}
+
 const SimulatorActionBar = (props: SimulatorActionBarProps) => {
   const device = useSimulator();
   const fs = useFileSystem();
+  const [simState, setSimState] = useState<SimState>(SimState.STOPPED);
+  const syncStatus = useSyncStatus();
   const handlePlay = useCallback(async () => {
     device.flash(fs, {
       partial: true,
       progress: () => {},
     });
+    setSimState(SimState.RUNNING);
   }, [device, fs]);
+  const handleStop = useCallback(() => {
+    device.stop();
+    setSimState(SimState.STOPPED);
+  }, [device, setSimState]);
+  useEffect(() => {
+    if (syncStatus === SyncStatus.OUT_OF_SYNC) {
+      handleStop();
+    }
+  }, [handleStop, syncStatus]);
   const size = "md";
   return (
     <HStack {...props} justifyContent="center" spacing={2.5} py={2} px={1}>
-      <IconButton
-        size={size}
-        variant="solid"
-        onClick={handlePlay}
-        icon={<RiPlayFill />}
-        aria-label="Run"
-      />
+      {syncStatus === SyncStatus.OUT_OF_SYNC ||
+      simState === SimState.STOPPED ? (
+        <IconButton
+          size={size}
+          variant="solid"
```

Comment:
Maybe we should drop this now? None of the designs have had a primary button here.
If it ends up just being stop that's an even stronger case for it not being primary.

Reply:
Agreed and aligned with the designs - [`bac8a8c`](https://github.com/microbit-foundation/python-editor-next/pull/858/commits/bac8a8c6b016740b7997e76b1a2c105e9a2a6d9e).